operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -263,9 +263,9 @@ private void handleResult(Reconciliation reconciliation, AsyncResult<Void> resul
         } else {
             Throwable cause = result.cause();
             if (cause instanceof InvalidConfigParameterException) {
-                log.warn("{}: Failed to reconcile {}", reconciliation, cause.getMessage());
+                log.error("{}: Failed to reconcile {}", reconciliation, cause.getMessage());
             } else if (!(cause instanceof UnableToAcquireLockException)) {
-                log.warn("{}: Failed to reconcile", reconciliation, cause);
+                log.error("{}: Failed to reconcile", reconciliation, cause);
```

Comment:
Elsewhere we've used (OK, _I_'ve used) error to mean "the operator itself is in a broken state". An invalid CR is not really an error: we expect users to make mistakes, and when they do nothing is broken, they just need to correct the mistake. 

Now, we can agree to change that policy, but I think we should 

* Have a discussion about it
* Have a PR to consistently find all the warnings which should now be errors.

Reply:
Hi, just thought I would put my two cents in...

I think that Strimzi (and indeed all operators) should use warnings for an invalid CR. A lot of people who run cloud applications in production may be filtering logs for errors and I should imagine these errors could get to be quite frequent under certain conditions.

I think it isn't the job of an operator to report an error for an invalid CR, just like how you probably wouldn't _usually_  log an error if someone made an invalid request to an API. The Strimzi operator should report an error only if the operator itself errors.

In an ideal world I think invalid CRs would be alerted to their own invalid state by a status being set inside them by the operator, though this is probably non-trivial to implement.

I am happy to be persuaded otherwise though :)